test/Database/LSMTree/Model/Session.hs

```diff
@@ -575,9 +586,11 @@ openSnapshot label name = do
     case Map.lookup name snaps of
       Nothing ->
         throwError ErrSnapshotDoesNotExist
-      Just (Snapshot conf label' tbl) -> do
+      Just (Snapshot conf label' tbl corrupted) -> do
         when (label /= label') $
           throwError ErrSnapshotWrongType
+        when corrupted $
+          throwError ErrSnapshotCorrupted
```

Comment:
I think the order should also be the other way around. The SUT will not be able to compare labels until it has read/deserialised/checksummed all the snapshot files